lib/srv/db/mongodb/engine.go

```diff
@@ -87,6 +88,20 @@ func (e *Engine) HandleConnection(ctx context.Context, sessionCtx *common.Sessio
 	defer closeFn()
 	e.Audit.OnSessionStart(e.Context, sessionCtx, nil)
 	defer e.Audit.OnSessionEnd(e.Context, sessionCtx)
+
+	// Create a session tracker so that other services, such as
+	// the session upload completer, can track the session's lifetime.
+	cancelCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	if err := sessionCtx.TrackSession(cancelCtx, e.EngineConfig); err != nil {
+		return trace.Wrap(err)
+	}
+	defer func() {
+		if err := services.UpdateSessionTrackerState(ctx, e.EngineConfig.AuthClient, sessionCtx.ID, types.SessionState_SessionStateTerminated); err != nil {
+			e.EngineConfig.Log.WithError(err).Warningf("Failed to update session tracker state for session %v.", sessionCtx.ID)
+		}
+	}()
```

Comment:
Instead of duplicating this in every database engine, can this be done in some common place? E.g. somewhere in Server or streamer.go?

Reply:
From what I can tell, in order to capture the lifetime of the db session, it has to be called from within the engine (same as audit start/end). I've trimmed some code down so it isn't so cumbersome.